WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1550,10 +1551,17 @@ private void showGutenbergInformativeDialog() {
     }
 
     private void setGutenbergEnabledIfNeeded() {
+        boolean newSitePopupNeeded = AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl());
         if ((TextUtils.isEmpty(mSite.getMobileEditor()) && !mIsNewPost)
-            || AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl())) {
+            || newSitePopupNeeded) {
             SiteUtils.enableBlockEditor(mDispatcher, mSite);
-            AnalyticsUtils.trackWithSiteDetails(Stat.EDITOR_GUTENBERG_ENABLED, mSite);
+            Map<String, Object> properties = new HashMap<>();
```

Comment:
Nit but, can we move the whole line `Map<String, Object> properties = new HashMap<>();` inside the conditional to be closer to the code using it? We don't need it outside, right?

While at it, and certainly not a blocker, WDYT about introducing a helper method to the enum object to quickly return a prefilled properties map? Something like:

```
public enum BlockEditorEnabledSource {
    VIA_SITE_SETTINGS,
    ON_SITE_CREATION,
    ON_BLOCK_POST_OPENING;

    public Map<String, Object> asPropertyMap() {
        Map<String, Object> properties = new HashMap<>();
        properties.put("source", name());
        return properties;
    }
}
```

We can then for example use it as `BlockEditorEnabledSource.ON_BLOCK_POST_OPENING.asPropertyMap()`.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/10300/commits/5809aa71e382401a7b9a19bb5fc277d2f51bac93